weave/tests/trace/test_weave_client.py

```diff
@@ -1200,6 +1206,100 @@ def test_table_partitioning(network_proxy_client):
     )
 
 
+def test_call_batch_size_handling(network_proxy_client, caplog):
```

Comment:
it is worth putting a similar test at the client level to ensure `remote_client._strip_large_values_from_batch(` gets called at the right spot